src/engine/Operation.h

```diff
@@ -44,25 +44,27 @@ class Operation {
       // in scope
       try {
         computeResult(newResult.get());
-      } catch (const ad_semsearch::Exception& e) {
-        if (e.getErrorCode() != ad_semsearch::Exception::QUERY_ABORTED) {
-          // Only print the Operation at the innermost failure
-          LOG(ERROR) << "Failed to compute Operation result for:" << endl;
-          LOG(ERROR) << asString() << endl;
-          LOG(ERROR) << e.getFullErrorMessage() << endl;
-        }
+      } catch (const ad_semsearch::AbortException& e) {
+        // AbortExceptions have already been printed simply rethrow to
+        // unwind the callstack until the whole query is aborted
+        throw;
+      } catch (const std::exception& e) {
+        // Only print the Operation at the innermost (original) failure
+        // then "rethrow" as special ad_semsearch::AbortException
+        LOG(ERROR) << "Failed to compute Operation result for:" << endl;
+        LOG(ERROR) << asString() << endl;
+        LOG(ERROR) << e.what() << endl;
         newResult->abort();
         // Rethrow as QUERY_ABORTED allowing us to print the Operation
         // only at innermost failure of a recursive call
-        throw ad_semsearch::Exception(ad_semsearch::Exception::QUERY_ABORTED,
-                                      e.getErrorDetails());
+        throw ad_semsearch::AbortException(e);
       }
```

Comment:
I still think a catch all block in the form of `catch(...) {` would be a good idea here, as we might otherwise run into deadlocks when weird exceptions are being thrown (as pretty much anything could be used as an exception object, and exceptions do not have to inherit from std::exception). With our current code all exceptions should inherit from std::exception, so this is not a critical problem right now, but might help with otherwise unexpected behavior in the future.

Reply:
Hmm, then I wouldn't have an object to pass to the `AbortException` constructor so I would need that to be an extra block. Let's see how that looks

Reply:
According to [cppreference](https://en.cppreference.com/w/cpp/error/exception) all standard library exceptions inherit from `std::exception` so I'm not sure if this warrants uglier code. If there's an exception that isn't from the standard library or `ad_semsearch::Exception` something would be pretty fishy

Reply:
I am fine with both solutions. We generally do not use third-party libraries in this part, so the `catch(...)` is not necessary for me. But it does not bother me too much, since it is very explicit and readable what happens here.